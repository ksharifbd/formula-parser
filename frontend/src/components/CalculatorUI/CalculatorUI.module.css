.CalculatorUI {
    display: flex;
    flex-direction: row;
}

.CalculatorUI__Sequence {
    margin-right: 1rem;
}
