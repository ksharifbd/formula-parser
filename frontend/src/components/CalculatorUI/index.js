import React from 'react';
import PropTypes from 'prop-types';

import MediaText from '../MediaText';
import UserInput from '../UserInput';
import classes from './CalculatorUI.module.css';

const CalculatorUI = ({
    sequence,
    formula,
    variables,
    value,
    onChange,
    result,
}) => (
    <li className={classes.CalculatorUI}>
        <p className={classes.CalculatorUI__Sequence}>{sequence}.</p>
        <div>
            <MediaText
                marginRight='.8rem'
                left={<p>Formula:</p>}
                right={<p>{formula}</p>} />
            <p>Inputs:</p>
            {variables.map(variable => (
                <UserInput
                    key={variable}
                    label={`${variable}:`}
                    value={value}
                    onChange={onChange} />
            ))}
            <MediaText
                marginRight='2rem'
                left={<p>Result:</p>}
                right={<p>{result}</p>} />
        </div>
    </li>
);

CalculatorUI.propTypes = {
    sequence: PropTypes.number.isRequired,
    formula: PropTypes.string.isRequired,
    variables: PropTypes.arrayOf(PropTypes.string).isRequired,
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    result: PropTypes.number.isRequired,
};

export default CalculatorUI;
