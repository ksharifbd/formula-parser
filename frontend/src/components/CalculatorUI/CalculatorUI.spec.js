import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import CalculatorUI from './';

describe('Calculator UI', () => {
    const tree = shallow(
        <CalculatorUI
            sequence={1}
            formula='a+b'
            result={5}>
                {['a', 'b'].map(el => <span key={el}>{el}</span>)}
            </CalculatorUI>
    );

    it('should render correctly ', () => {
        expect(toJson(tree)).toMatchSnapshot();
    });
});
