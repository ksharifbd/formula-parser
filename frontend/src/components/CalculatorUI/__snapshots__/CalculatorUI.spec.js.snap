// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator UI should render correctly  1`] = `
<li
  className="CalculatorUI"
>
  <p
    className="CalculatorUI__Sequence"
  >
    1
    .
  </p>
  <div>
    <MediaText
      left={
        <p>
          Formula:
        </p>
      }
      marginRight=".8rem"
      right={
        <p>
          a+b
        </p>
      }
    />
    <p>
      Inputs:
    </p>
    <span
      key="a"
    >
      a
    </span>
    <span
      key="b"
    >
      b
    </span>
    <MediaText
      left={
        <p>
          Result:
        </p>
      }
      marginRight="2rem"
      right={
        <p>
          5
        </p>
      }
    />
  </div>
</li>
`;
